cql3/term.hh

```diff
@@ -45,13 +45,64 @@
 #include "cql3/query_options.hh"
 #include "cql3/values.hh"
 #include "cql3/cql_value.hh"
+#include "cql3/functions/scalar_function.hh"
 
 namespace cql3 {
 
 class term;
 class terminal;
 class prepare_context;
 
+struct delayed_marker_value;
+struct delayed_tuple_value;
+struct delayed_list_value;
+struct delayed_set_value;
+struct delayed_map_value;
+struct delayed_user_type_value;
+struct delayed_function_value;
+
+using delayed_cql_value = std::variant<
+    delayed_marker_value,
+    delayed_tuple_value,
+    delayed_list_value,
+    delayed_set_value,
+    delayed_map_value,
+    delayed_user_type_value,
+    delayed_function_value>;
```

Comment:
I think these are redundant with what we have (after my series converting term::raw). For example, delayed_tuple_value is tuple_constructor{{...}}, where at least some of the elements are bind_variables or contain them indirectly.

Of course, we can also say that my series is redundant wrt. this change, but I think my series brings us closer to more generic expressions. Again looking at tuples, tuple_constructor can take any expression for its elements, not just term/new_term. So you can have a tuple_constructor{column_value, bind_variable, constant} which is not expressible with new_term.

Reply:
You are right, I'm still going through the series, will remove.